File: python/src/models/patient_profile.py
"""
Patient Profile Data Model
Represents comprehensive patient information including demographics,
accessibility needs, and care context.
"""

from datetime import datetime
from enum import Enum
from typing import Optional, List
from pydantic import BaseModel, Field


class HealthLiteracyLevel(str, Enum):
    """Health literacy level of the patient"""
    BASIC = "basic"
    INTERMEDIATE = "intermediate"
    ADVANCED = "advanced"


class Demographics(BaseModel):
    """Patient demographic information"""
    age: int = Field(..., ge=0, le=150)
    gender: Optional[str] = None
    preferred_name: Optional[str] = None


class AccessibilitySettings(BaseModel):
    """Accessibility preferences and requirements"""
    screen_reader_enabled: bool = False
    high_contrast_mode: bool = False
    font_size: str = Field(default="medium", pattern="^(small|medium|large|extra-large)$")
    voice_output_enabled: bool = False
    keyboard_navigation_only: bool = False


class LanguageSettings(BaseModel):
    """Language preferences for patient communication"""
    primary_language: str
    secondary_languages: Optional[List[str]] = None
    translation_enabled: bool = False


class NotificationPreferences(BaseModel):
    """Notification channel preferences"""
    email: bool = False
    sms: bool = False
    push: bool = False


class CommunicationPreferences(BaseModel):
    """Patient communication preferences"""
    preferred_input_mode: str = Field(default="text", pattern="^(text|voice|mixed)$")
    preferred_output_mode: str = Field(default="text", pattern="^(text|audio|visual|mixed)$")
    notification_preferences: NotificationPreferences
    reminder_frequency: str = Field(default="medium", pattern="^(low|medium|high)$")


class PatientProfile(BaseModel):
    """Complete patient profile with all relevant information"""
    patient_id: str
    demographics: Demographics
    accessibility_needs: AccessibilitySettings
    language_preferences: LanguageSettings
    health_literacy_level: HealthLiteracyLevel
    communication_preferences: CommunicationPreferences
    # care_context is defined in care_context.py to avoid circular imports

File: src/shared/types/patient-profile.ts
/**
 * Patient Profile Data Model
 * Represents comprehensive patient information including demographics,
 * accessibility needs, and care context.
 */

export enum HealthLiteracyLevel {
  BASIC = 'basic',
  INTERMEDIATE = 'intermediate',
  ADVANCED = 'advanced',
}

export interface Demographics {
  age: number;
  gender?: string;
  preferredName?: string;
}

export interface AccessibilitySettings {
  screenReaderEnabled: boolean;
  highContrastMode: boolean;
  fontSize: 'small' | 'medium' | 'large' | 'extra-large';
  voiceOutputEnabled: boolean;
  keyboardNavigationOnly: boolean;
}

export interface LanguageSettings {
  primaryLanguage: string;
  secondaryLanguages?: string[];
  translationEnabled: boolean;
}

export interface CommunicationPreferences {
  preferredInputMode: 'text' | 'voice' | 'mixed';
  preferredOutputMode: 'text' | 'audio' | 'visual' | 'mixed';
  notificationPreferences: {
    email: boolean;
    sms: boolean;
    push: boolean;
  };
  reminderFrequency: 'low' | 'medium' | 'high';
}

export interface PatientProfile {
  patientId: string;
  demographics: Demographics;
  accessibilityNeeds: AccessibilitySettings;
  languagePreferences: LanguageSettings;
  healthLiteracyLevel: HealthLiteracyLevel;
  communicationPreferences: CommunicationPreferences;
  careContext: CareContext;
}

// Forward declaration - CareContext is defined in care-context.ts
export interface CareContext {
  dischargeDate: Date;
  dischargeDiagnosis: string[];
  careInstructions: CareInstruction[];
  medications: Medication[];
  followUpAppointments: Appointment[];
  careTeam: CareTeamMember[];
  emergencyContacts: Contact[];
}

export interface CareInstruction {
  instructionId: string;
  category: 'medication' | 'activity' | 'diet' | 'wound-care' | 'monitoring' | 'other';
  description: string;
  priority: 'urgent' | 'high' | 'medium' | 'low';
  frequency?: string;
  duration?: string;
  startDate?: Date;
  endDate?: Date;
}

export interface Medication {
  medicationId: string;
  name: string;
  dosage: string;
  frequency: string;
  route: string;
  purpose: string;
  startDate: Date;
  endDate?: Date;
  instructions: string;
  sideEffects?: string[];
  interactions?: string[];
}

export interface Appointment {
  appointmentId: string;
  provider: string;
  specialty: string;
  date: Date;
  location: string;
  purpose: string;
  preparationInstructions?: string[];
}

export interface CareTeamMember {
  memberId: string;
  name: string;
  role: string;
  specialty?: string;
  contactInfo: Contact;
}

export interface Contact {
  name: string;
  relationship?: string;
  phone: string;
  email?: string;
  preferredContactMethod: 'phone' | 'email' | 'sms';
}
